### PR TITLE
PEO-429 Update fasterer config runner

### PR DESCRIPTION
**Pull request by rostikkkk2**
[PEO-429](https://garage.atlassian.net/browse/PEO-429)

### Description

Need to provide next ability:
load a configuration file form the application if it exists
otherwise, load default configuration file (from the config/fasterer/config.yml)

### Before submitting the merge request make sure the following are checked

- [ ] Followed the style guidelines of this project
- [ ] Performed a self-review of own code
- [ ] Wrote the tests that prove that fix is effective/that feature works

### CODE REVIEW

File: lib/inquisition/railtie.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
 module Inquisition
-  class Railtie < ::Rails::Railtie
-  end
+  # class Railtie < ::Rails::Railtie
```
Comment (chubchenko) [inline lib/inquisition/railtie.rb]: ?

File: lib/inquisition/fasterer/config.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+module Inquisition
+  module Fasterer
+    class Config < ::Fasterer::Config
+      def nil_file
+        { SPEEDUPS_KEY => load_config['speedups'], EXCLUDE_PATHS_KEY => load_config['exclude_paths'] }
+      end
+
+      def load_config
+        YAML.load_file(File.join(__dir__, '../../../config/fasterer/.fasterer.yml'))
```
Comment (chubchenko) [inline lib/inquisition/fasterer/config.rb]: Use `Inquisition.root`

File: lib/inquisition/fasterer/config.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+require 'yaml'
+
+module Inquisition
+  module Fasterer
+    class Config < ::Fasterer::Config
+      def nil_file
```
Comment (chubchenko) [inline lib/inquisition/fasterer/config.rb:6]: Strange name

File: spec/inquisition/fasterer/config_spec.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
+RSpec.describe Inquisition::Fasterer::Config do
+  describe '#nil_file' do
+    subject(:run_method) { described_class.new.nil_file }
+
+    let(:load_config) do
+      { 'speedups' => {}, 'exclude_paths' => {} }
+    end
+
+    it 'check upper class' do
+      expect(described_class).to be < ::Fasterer::Config
```
Comment (chubchenko) [inline spec/inquisition/fasterer/config_spec.rb]: Not sure that this test related to the `nil_file` method

File: spec/inquisition/fasterer/config_spec.rb [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    end
+
+    context 'when call method' do
```
Comment (chubchenko) [inline spec/inquisition/fasterer/config_spec.rb]: Strange context

File: spec/inquisition/fasterer/config_spec.rb [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+      before { allow(YAML).to receive(:load_file).and_return(load_config) }
+
+      it 'return hash' do
```
Comment (chubchenko) [inline spec/inquisition/fasterer/config_spec.rb]: `returns` instead of `return`

File: spec/inquisition/fasterer/config_spec.rb [hunk 7]
```diff
@@ -0,0 +1,35 @@
+RSpec.describe Inquisition::Fasterer::Config do
+  describe '#nil_file' do
+    subject(:run_method) { described_class.new.nil_file }
```
Comment (chubchenko) [inline spec/inquisition/fasterer/config_spec.rb]: strange name of subject

File: spec/inquisition/fasterer/config_spec.rb [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+      before { allow(YAML).to receive(:load_file).and_return(load_config) }
+
+      it 'return hash' do
+        expect(run_method).to eq(load_config)
+      end
+    end
+  end
+
+  describe '#load_config' do
+    subject(:run_method) { described_class.new.load_config }
```
Comment (chubchenko) [inline spec/inquisition/fasterer/config_spec.rb]: Same

File: spec/inquisition/fasterer/config_spec.rb [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+    it 'check upper class' do
+      expect(described_class).to be < ::Fasterer::Config
```
Comment (chubchenko) [inline spec/inquisition/fasterer/config_spec.rb]: same

File: lib/inquisition/fasterer/config.rb [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+        { SPEEDUPS_KEY => load_config['speedups'], EXCLUDE_PATHS_KEY => load_config['exclude_paths'] }
+      end
+
+      def load_config
```
Comment (chubchenko) [inline lib/inquisition/fasterer/config.rb]: No sure that this method should be public

File: lib/inquisition/fasterer/file_traverser.rb [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+
+module Inquisition
+  module Fasterer
+    class FileTraverser < ::Fasterer::FileTraverser
+      def initialize(path)
+        super
+        @config = Config.new
+      end
+
+      def offenses_grouped_by_type(analyzer)
```
Comment (chubchenko) [inline lib/inquisition/fasterer/file_traverser.rb:12]: Why do we need this method?

File: lib/inquisition/fasterer/runner.rb [hunk 12]
```diff
@@ -14,6 +15,12 @@ def call
       def scan_file(path)
         analyzer = ::Fasterer::Analyzer.new(path)
         analyzer.scan
+        check_analyser_config_rules(analyzer)
+      end
+
+      def check_analyser_config_rules(analyzer)
+        return unless @fasterer.offenses_grouped_by_type(analyzer).any?
```
Comment (chubchenko) [inline lib/inquisition/fasterer/runner.rb:22]: Why do we need this check?

**Review by chubchenko — CHANGES_REQUESTED** [review]

File: lib/inquisition/fasterer/file_traverser.rb (hunk 11 above)
Comment (rostikkkk2) [reply lib/inquisition/fasterer/file_traverser.rb:12]: I need to use this method in runner, but in `::Fasterer::FileTraverser` it is `private`
we need this to check config file include type error that we have

File: lib/inquisition/fasterer/runner.rb (hunk 12 above)
Comment (rostikkkk2) [reply lib/inquisition/fasterer/runner.rb:22]: it is check type of analyzer.errors and show, contain this type error in config file or not. if not contain in config this type, this mehtod returne current error, if contain, I mush not create issue with this type error

File: lib/inquisition/fasterer/config.rb (hunk 3 above)
Comment (rostikkkk2) [reply lib/inquisition/fasterer/config.rb:6]: I can't rename this because this metod call from `::Fasterer::Config`

**Pull request merged**